test/apps/iodemo/io_demo.cc

```diff
@@ -439,6 +444,9 @@ class P2pDemoCommon : public UcxContext {
             case UCS_MEMORY_TYPE_HOST:
                 buffer = UcxContext::memalign(ALIGNMENT, size,
                                               pool.name().c_str());
+                if (context != NULL && buffer != NULL) {
+                    assert(context->map_buffer(size, buffer, &memh, 0) == UCS_OK);
```

Comment:
if !UCS_OK, should UcxContext::free(buffer)

Reply:
if !UCS_OK, assert would call std::abort() which would reclaim all dynamic allocated memory when exiting the program i assume.

Reply:
@huaxiangfan But assert is a macro, you can't make sure it will always be defined as std::abort(). rgiht?

Reply:
right

Reply:
Add also, please don't call function which must be called in the parameter of assert(), this is not safe. For example if assert is defined as blank macro, then the calling of the function would be missed.